src/AddImageChecks.cpp

```diff
@@ -478,8 +479,23 @@ Stmt add_image_checks(Stmt s,
             // Check the var passed in equals the constrained version (when not in inference mode)
             asserts_constrained.push_back(AssertStmt::make(var == constrained_var, error));
         }
+        if (param.defined() && param.host_alignment() > 1) {
+            string host_name = name + ".host";
+            int alignment_required = param.host_alignment();
+            Expr host_ptr = Variable::make(Int(32), host_name);
```

Comment:
I think this variable should be of type `Handle()`, and then the and operation below should be (cast<uint64_t>(host_ptr) & (alignment_required - 1)). Then you don't need to make the Mod op handle pointers in CodeGen_LLVM, and it's conceptually a bit cleaner (IMO).